includes/core/server/class-request.php

```diff
@@ -369,6 +379,26 @@ protected function set_request_route_id_parameter( $id ) {
 			}
 		}
 	}
+	/**
+	 * Assign the WordPress filter hook and the arguments specific to the previously assigned hook.
+	 *
+	 * @param string $hook WordPress filter hook name.
+	 * @param mixed  ...$args Arguments for the filter hook.
+	 */
+	public function apply_hook( string $hook, ...$args ) {
```

Comment:
Maybe change this to something like `assign_hook`? `apply_hook` sounds like something that would be executed immediately, and that's not the case here.

Also, the description arguments says "filter hook", which is wrong. I think "filter/hook" or just "hook" would be better.

Reply:
Furthermore, considering that `set_hook_args` is a separate method, let's not try to combine things within this one, and let it focus on the hook name. Otherwise we might have a silent side-effect of `apply_hook( 'some_hook' )` cleaning up args without a warning.

Reply:
Good point. I refactored the method signature by making only filter or hook name as an argument and changed the name of the function to assign_hook.

Addressed here: https://github.com/Automattic/woocommerce-payments/pull/7314/commits/b72ea986fdd961f8586f2173aa7c8bf0087cbeb8